src/utils/uploader.js

```diff
@@ -112,18 +112,35 @@ export class BlobUploader {
         tx.blobVersionedHashes = versionedHashes;
         tx.blobs = ethersBlobs;
         tx.kzg = kzg;
-        return await this.#wallet.sendTransaction(tx);
+        return tx;
     }
 
-    async sendTxLock(tx, blobs) {
+    async sendTx(tx, blobs) {
+        const txObj = await this.#populate(tx, blobs);
+        return await this.#wallet.sendTransaction(txObj);
+    }
+
+    async #sendTxLock(tx) {
         const release = await this.#mutex.acquire();
```

Comment:
Which part does mutex protect?

Reply:
Ensure that the acquisition of nonce and the broadcast of tx are incremented in order.